### PR TITLE
Update ability base cooldown times and scalars.

### LINKED ISSUE #104: Ability cool downs don't match in game cool down
Cool down estimates in d2ap don't match the cool downs displayed in game across the different tiers. 

Example: Warlock fusion grenade at Discipline T7 in game shows 0:52 and d2ap shows 0:40

I know there have been some buffs and nerfs to cool downs so I thought I'd create this issue to help keep track of the progress. I started converting some of the in-game cool downs to a ratio and will try to contribute to this if I'm able to make some time to work on it.

### PR DESCRIPTION
**Pull request by THATDONFC**
Adjusted ability cooldowns and scalars to reflect current in game cooldown timers at all tiers
Adjusted resilience scale
Added Strand abilities
Added Phoenix Dive

Fixes #104

### CODE REVIEW

**Comment by THATDONFC** [comment]
Hold off on merging this one. Looks like it was more than just a few of the warlock melee cooldowns that were off in the clarity repo. I'll update those soon and push here.

**Comment by THATDONFC** [comment]
I adjusted the melee cooldowns to match the [most recent commit](https://github.com/Database-Clarity/Character-Stats/commit/75967a7d909e1ca665703f788b2de089f9682e0d) to the clarity repo. The 90 second melee tier no longer exists so it was removed.

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 1]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
 
 export const MeleeAbilitiesPerClassAndTier: EnumDictionary<CharacterClass, string[]> = {
   [CharacterClass.Titan]: [
+    "", // 2:17
+    "", // 2:05
+    "Ballistic Slam, Shield Bash, Frenzied Blade", // 1:54
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:8]: 114
Ballistic Slam
image
Shield Bash
image

Frenzied Blade
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 2]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
     "Shiver Strike", // 1:53
-    "", // 1:49
-    "Frontal Assault", // 1:46
-    "Shield Throw", // 1:40
-    "Throwing Hammer, Hammer Strike, Shield Bash, Seismic Strike, Ballistic Slam, Thunderclap", // 1:30
+    "", // 1:52
+    "", // 1:51
+    "Hammer Strike, Seismic Strike", // 1:41
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:12]: 101
Hammer Strike
image

Seismic Strike
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 3]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:40
+    "Shield Throw, Throwing Hammer, Thunderclap", // 1:31
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:14]: 91
Shield Throw
image

Throwing Hammer
image

Thunderclap
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 4]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
-    "", // 0:15
+    "", // 0:40
   ],
   [CharacterClass.Hunter]: [
-    "Withering Blade", // 1:53
-    "Weighted Throwing Knife", // 1:49
-    "", // 1:46
-    "Proximity Expl. Knife, Tempest Strike, Disorienting Blow", // 1:40
-    "Lightweight Knife, Snare Bomb", // 1:30
+    "Weighted Throwing Knife", // 2:17
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:20]: 137
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 5]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "Threaded Spike", // 2:05
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:21]: 125
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 6]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:54
+    "", // 1:53
+    "", // 1:52
+    "Proximity Expl. Knife", // 1:51
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:25]: 111
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 7]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:41
+    "Lightweight Knife, Withering Blade, Disorienting Blow", // 1:40
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:27]: 100
image

image
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 8]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "Snare Bomb, Tempest Strike", // 1:31
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:28]: 91
image

image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 9]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:23
     "Knife Trick", // 1:22
-    "Combination Blow", // 0:15
+    "Combination Blow", // 0:40
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:31]: image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 10]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
   ],
   [CharacterClass.Warlock]: [
-    "Penumbral Blast, Ball Lightning", // 1:53
-    "", // 1:49
-    "", // 1:46
-    "Celestial Fire", // 1:40
-    "Incinerator Snap, Pocket Singularity, Chain Lightning, Rising Storm", // 1:30
+    "", // 2:17
+    "", // 2:05
+    "Ball Lightning", // 1:54
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:36]: 114
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 11]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:53
+    "Celestial Fire", // 1:52
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:38]: 112
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 12]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:51
+    "Penumbral Blast", // 1:41
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:40]: 101
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 13]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "", // 1:40
+    "Pocket Singularity, Chain Lightning, Arcane Needle", // 1:31
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:42]: 91
image
image
image

File: src/app/data/cooldowns/cooldowns_melee.ts [hunk 14]
```diff
@@ -3,31 +3,46 @@ import {CharacterClass} from "../enum/character-Class";
...
+    "Incinerator Snap", // 1:23
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldowns_melee.ts:43]: 83
image

File: src/app/data/cooldowns/cooldowns_super.ts [hunk 15]
```diff
@@ -3,23 +3,20 @@ import {CharacterClass} from "../enum/character-Class";
...
   [CharacterClass.Hunter]: [
-    "Spectral Blades",// 10:25
-    "Arc Staff, Golden Gun", // 9:16
-    "Morbius Quiver", // 8:20
+    "Spectral Blades, Golden Gun, Arc Staff, Silk Strike", // 9:16
+    "Moebius Quiver, Gathering Storm", // 8:20
     "Deadfall, Blade Barrage, Silence & Squall", // 7:35
     "", // 6:57
   ],
   [CharacterClass.Warlock]: [
```
[the remark this answers is not in the document]
Comment (TheYeshir13) [reply src/app/data/cooldowns/cooldowns_super.ts:17]: Chaos Reach is wrong in my table - rechecked with clarity
image

**Review by TheYeshir13 — APPROVED** [review]
cooldown_grenade.ts, cooldown_meele.ts,cooldown_super.ts can be approved

cooldown_definitions and stat-cooldown-tooltip need further testing

File: src/app/data/cooldowns/cooldown_definitions.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
-export const SUPER_COOLDOWN_RATIO_PER_TIER = [1.44, 1.276, 1.144, 1.0, 0.9488, 0.9024, 0.8608, 0.8224, 0.798, 0.776, 0.7664]
-export const ABILITY_COOLDOWN_RATIO_PER_TIER = [1.25, 1.14, 1.04, 1.0, 0.83, 0.72, 0.62, 0.55, 0.50, 0.455, 0.39]
-export const CLASS_COOLDOWN_RATIO_PER_TIER = [1.427, 1.256, 1.11, 1.0, 0.91461, 0.829, 0.7683, 0.72, 0.622, 0.561, 0.5]
+export const SUPER_COOLDOWN_RATIO_PER_TIER = [1.439, 1.276, 1.144, 1.0, 0.9478, 0.90, 0.8598, 0.8216, 0.798, 0.7751, 0.7651]
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldown_definitions.ts:1]: image
As discussed on discord, its close enough

File: src/app/data/cooldowns/cooldown_definitions.ts [hunk 17]
```diff
@@ -1,6 +1,6 @@
...
+export const ABILITY_COOLDOWN_RATIO_PER_TIER = [1.25, 1.131, 1.039, 1.0, 0.905, 0.831, 0.766, 0.71, 0.642, 0.569, 0.498]
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldown_definitions.ts:2]: image
Same as with super, close enough

File: src/app/components/authenticated-v2/overlays/stat-cooldown-tooltip/stat-cooldown-tooltip.component.ts [hunk 18]
```diff
@@ -125,13 +125,19 @@ const CooldownInfo: EnumDictionary<ArmorStat, CooldownEntry[]> = {
       benefitDirection: BenefitDirection.Negative,
       valueFunction: t => 82 * CLASS_COOLDOWN_RATIO_PER_TIER[t],
       formatting: (t) => reformatTimeMMMSS(t)
+    },
+    {
+      name: "Phoenix Dive",
+      benefitDirection: BenefitDirection.Negative,
+      valueFunction: t => 55 * CLASS_COOLDOWN_RATIO_PER_TIER[t],
```
Comment (TheYeshir13) [inline src/app/components/authenticated-v2/overlays/stat-cooldown-tooltip/stat-cooldown-tooltip.component.ts:132]: image

File: src/app/data/cooldowns/cooldown_definitions.ts [hunk 19]
```diff
@@ -1,6 +1,6 @@
...
+export const CLASS_COOLDOWN_RATIO_PER_TIER = [1.428, 1.25, 1.111, 1.0, 0.909, 0.8333, 0.7692, 0.7142, 0.666, 0.625, 0.5882]
```
Comment (TheYeshir13) [inline src/app/data/cooldowns/cooldown_definitions.ts:3]: image

According to Clarity, Hunter Dodge does have a different Scalar

File: src/app/components/authenticated-v2/overlays/stat-cooldown-tooltip/stat-cooldown-tooltip.component.ts
Comment (TheYeshir13) [inline src/app/components/authenticated-v2/overlays/stat-cooldown-tooltip/stat-cooldown-tooltip.component.ts:1]: Base Values were confirmed in the cooldown files

**Review by TheYeshir13 — APPROVED** [review]
Overall approved.
As discussed there are slight deviations in the scalars (compared to clarity) but they should be fine.

Hunter Dodges have a different scalar than the other class abilities, this might be addressed in a further issue

Also might wanna change Warlock Super to Winters Wrath

And add a disclaimer similar to clarity "Cooldown times might be ~1 second off in some instances due to how these numbers are calculated"

**Comment by Mijago** [comment]
I will merge it as soon as #133 is done.

**Comment by TheYeshir13** [comment]
Maybe this need to be adjusted with the changes for #133

**Comment by Mijago** [comment]
Closing in favor of  https://github.com/Mijago/D2ArmorPicker/pull/157

**Pull request closed without merging**